### PR TITLE
Supress plugin discovery from plugins, which is redundant and noisy in the logs

### PR DESCRIPTION
**Pull request by cbednarski**
This one has been bothering me for a bit. Copying more or less the same fix I applied in Terraform.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -46,6 +47,12 @@ func decodeConfig(r io.Reader, c *config) error {
 // Hence, the priority order is the reverse of the search order - i.e., the
 // CWD has the highest priority.
 func (c *config) Discover() error {
+	// If we are already inside a plugin process we should not need to
+	// discover anything.
+	if os.Getenv(plugin.MagicCookieKey) != "" {
```
Comment (mwhooker) [inline config.go]: should we check that it equals `$PACKER_PLUGIN_MAGIC_COOKIE`? like https://github.com/mitchellh/packer/blob/master/packer/plugin/server.go#L42

Comment (cbednarski) [inline config.go]: Hmm... These are not actually logically equivalent, but it probably won't matter.

Comment (cbednarski) [inline config.go]: This is updated, btw!

**Comment by rickard-von-essen** [comment]
By the way we should print a warning you override a built-in plugin.

**Pull request merged**